pkg/storaged/client.js

```diff
@@ -206,6 +214,66 @@
     return false;
 }
 
+export async function btrfs_poll() {
+    const usage_regex = /used\s+(?<used>\d+)\s+path\s+(?<device>[\w/]+)/;
+    if (!client.uuids_btrfs_subvols)
+        client.uuids_btrfs_subvols = { };
+    if (!client.uuids_btrfs_usage)
+        client.uuids_btrfs_usage = { };
+    if (!client.uuids_btrfs_volume)
+        return;
+
+    const uuids_subvols = { };
+    const uuids_usage = { };
+    for (const uuid of Object.keys(client.uuids_btrfs_volume)) {
+        const block = client.uuids_btrfs_blocks[uuid][0];
+        const block_fsys = client.blocks_fsys[block.path];
+        const mp = block_fsys.MountPoints[0];
+        if (mp) {
+            const output = await cockpit.spawn(["btrfs", "subvolume", "list", "-ap", utils.decode_filename(mp)], { superuser: true, err: "message" });
+            const subvols = [{ pathname: "/", id: 5 }];
+            for (const line of output.split("\n")) {
+                const m = line.match(/ID (\d+).*parent (\d+).*path (<FS_TREE>\/)?(.*)/);
+                if (m)
+                    subvols.push({ pathname: m[4], id: Number(m[1]) });
+            }
+            uuids_subvols[uuid] = subvols;
+
+            // HACK: this should be udisks, link udisks/libblockdev issue
+            // TODO: optimise into just parsing one `btrfs filesystem show`?
+            const usages = {};
+            const usage_output = await cockpit.spawn(["btrfs", "filesystem", "show", "--raw", uuid], { superuser: true, err: "message" });
+            for (const line of usage_output.split("\n")) {
+                const match = usage_regex.exec(line);
+                if (match) {
+                    const { used, device } = match.groups;
+                    usages[device] = used;
+                }
+            }
+            uuids_usage[uuid] = usages;
+        } else {
+            uuids_subvols[uuid] = null;
+            uuids_usage[uuid] = null;
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20231205-110900-7b3be2c9-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)